lib/extension.js

```diff
@@ -1,7 +1,7 @@
 var path = require('path');
 
 var EXTRE = /\.[^.]*/g;
-var LONGEXTRE = /^[.]?[^.]+([.].+[^.])$/;
+var LONGEXTRE = /^[.]?[^.]+([.].*[^.])$/;
```

Comment:
@sttk will this cause a ReDOS issue? Maybe it needs to be this:
```suggestion
var LONGEXTRE = /^[.]?[^.]+([.][^.]*[^.])$/;
```
![image](https://user-images.githubusercontent.com/992373/113786891-e6977300-96ee-11eb-8ce5-072f80efd206.png)

Do you agree?

Reply:
@phated That regexp does not pass tests.

```
$ node
> /^[.]?[^.]+([.][^.]*[^.])$/.test('aaa.js')
true
> /^[.]?[^.]+([.][^.]*[^.])$/.test('aaa.bbb.js')
false
> /^[.]?[^.]+([.][^.]*[^.])$/.test('aaa..bbb.js')
false
```

And I checked the regexp `/^[.]?[^.]+([.].*[^.])$/` with some tools, and the vulnerability was not detected.

- http://redos-checker.surge.sh/
- https://www.npmjs.com/package/safe-regex
- https://2bdenny.github.io/ReScue/

Reply:
@sttk I don't think those tools work correctly because I submitted the ReDOS RegExp from https://github.com/gulpjs/glob-parent/commit/f9231168b0041fea3f8f954b3cceb56269fc6366 and all the tools claim it is safe.

Reply:
@phated Because I don't know how to avoid ReDOS, I replaced regex processes in `extension.js` to alternative ways.

The following table is the benchmark result of `extension.js` before and after modification.
```
|                          | before            | after              |
|:-------------------------|------------------:|-------------------:|
| file.js                  | 3,465,168 ops/sec | 17,745,432 ops/sec |
| file.tmp.dot.js          | 1,453,870 ops/sec |  6,729,040 ops/sec |
| relative/path/to/file.js | 3,292,908 ops/sec | 14,813,461 ops/sec |

- Platform: Node.js 15.5.0 on Darwin 64-bit
- Machine: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz, 64GB
```